### PR TITLE
Added example for batch span processor

### PR DESCRIPTION
**Pull request by snehilchopra**
This PR contains a short example on how to use the *tracer* using a *batch span processor*.

Please let me know if a README.md would be helpful.

Thanks!

### CODE REVIEW

File: examples/batch/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (snehilchopra) [inline examples/batch/CMakeLists.txt:1]: I saw the cmake support was taken off in the simple example too. Should I add that to both?

File: examples/batch/main.cc [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#include <thread>
+
+constexpr int kNumSpans = 10;
+
+namespace
+{
+
+void initTracer()
+{
+  /*                         SET UP EXPORTER                                    */
```
Comment (pyohannes) [inline examples/batch/main.cc:17]: Please remove comments that don't add much to the information that is already given by the code. I think 
```c++
  auto exporter = std::unique_ptr<sdktrace::SpanExporter>(
      new opentelemetry::exporter::trace::OStreamSpanExporter);
```
makes it clear that a new exporter is set up.

File: examples/batch/main.cc [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  const std::chrono::milliseconds schedule_delay_millis = std::chrono::milliseconds(3000);
+
+  // We export `kNumSpans` after every `schedule_delay_millis` milliseconds.
+  const size_t max_export_batch_size = kNumSpans;
+
+  auto processor = std::shared_ptr<sdktrace::SpanProcessor>(new sdktrace::BatchSpanProcessor(
+      std::move(exporter), max_queue_size, schedule_delay_millis, max_export_batch_size));
+
+  // Wait for processor's worker thread to initialize.
+  std::this_thread::sleep_for(std::chrono::milliseconds(50));
```
Comment (pyohannes) [inline examples/batch/main.cc:37]: Why do we need this? Could we go without this?

File: examples/batch/main.cc [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+}  // namespace
+
+int main()
+{
+  // Removing this line will leave the default noop TracerProvider in place.
+  initTracer();
+
+  {
```
Comment (pyohannes) [inline examples/batch/main.cc:67]: Is there a reason for starting a new block?

File: examples/batch/main.cc [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    // Generate `kNumSpans` spans.
+    StartAndEndSpans();
+
+    printf("Exporting first batch of %d spans in 3 seconds ...\n", kNumSpans);
```
Comment (pyohannes) [inline examples/batch/main.cc:71]: Please use `cout` instead of `printf`.

File: examples/batch/main.cc [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  {
+    // Generate `kNumSpans` spans.
+    StartAndEndSpans();
+
+    printf("Exporting second batch of %d spans in 3 seconds ...\n", kNumSpans);
+    // Sleep for schedule_delay_millis + small delay to account for export time.
+    std::this_thread::sleep_for(std::chrono::milliseconds(3000 + 50));
+
+    // The spans should now be exported.
+    printf("....Exported!\n\n\n");
```
Comment (pyohannes) [inline examples/batch/main.cc:88]: You could make this more concise:

```c++
cout << " Creating second batch of " << kNumSpans << " spans and waiting 3 seconds ...\n";
StartAndEndSpans();
std::this_thread::sleep_for(std::chrono::milliseconds(3000 + 50));
cout << " ... Exported!\n\n\n";
```

File: examples/batch/main.cc (hunk 3 above)
Comment (snehilchopra) [reply examples/batch/main.cc:37]: Oops. Yes, we can. Now that we aren't checking `if(buffer_.size() < max_export_batch_size_` in the background routine of the batch processor, we can get rid of this.

Earlier, by the time this condition was checked in the worker thread, we already had `max_export_batch_size_` spans in the buffer, meaning that we didn't honor the `schedule_delay_millis_` time interval. Also, we'd not wait 3 seconds after printing out `Exporting first batch of 10 spans in 3 seconds ...\n` and would instead directly export span data to stdout, not giving the user a chance to see what's going on. Thus, I added the 50ms sleep.

Thank you for the heads up!

File: examples/batch/main.cc (hunk 4 above)
Comment (snehilchopra) [reply examples/batch/main.cc:67]: Nothing in particular, except for separating out the exports for more readability. 
No functional advantage.

I can drop them.

File: examples/batch/CMakeLists.txt (hunk 1 above)
Comment (pyohannes) [reply examples/batch/CMakeLists.txt:1]: Yes, let's have CMake support for both.

**Pull request merged**